Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector_test.py

```diff
@@ -0,0 +1,594 @@
+            "test12345",
+        ),
+    ],
+)
+def test_generate_id_for_event(
+    event_key: str, event_type: EventType, expected_results: str
+):
+    """
+    Given:
+        - event
+    When:
+        - run `generate_id_for_event` function
+    Then:
+        - Ensure that ID is returned which is generated
+          by concatenating values included in the event
+        - Ensure the function generates an ID even when
+          some of the values are missing or some are equal to `None`
+    """
+    dedup = Deduplicate([], event_type)
+    event = load_event_for_test(event_key)
+    result = dedup.generate_id_for_event(event[0])
+    assert result == expected_results
+
+
+@pytest.mark.parametrize(
+    "event_key, latest_time, expected_results",
+    [
+        (
+            "GET_EVENT_IDS_WITH_DUPLICATION_RISK",
+            "2023-07-15T10:00:18Z",
+            [
+                "<11111.11111.11111.1111@mx.test.com>",
+                "<22222.22222.22222.2222@mx.test.com>",
+            ],
+        )
+    ],
+)
+def test_get_event_ids_with_duplication_risk(
+    event_key: str, latest_time: str, expected_results: list[str]
+):
+    """
+    Given:
+        - The events
+    When:
+        - run `get_event_ids_with_duplication_risk` function
+    Then:
+        - Ensure the function returned the IDs of all events that are suspected of being duplicates
+    """
+    dedup = Deduplicate([], EventType.POLICY_LOGS)
+    events = load_event_for_test(event_key)
+    results = dedup.get_event_ids_with_duplication_risk(events, latest_time)
+    assert len(results) == len(expected_results)
+    for result in results:
+        assert result in expected_results
+
+
+@pytest.mark.parametrize(
+    "event_key, ids_fetched_by_type, time_from, expected_results",
+    [
+        (
+            "DEDUPLICATE_1",
+            ["<11111.11111.11111.1111@mx.test.com>"],
+            "2023-07-15T10:00:18Z",
+            2,
+        ),
+        (
+            "DEDUPLICATE_2",
+            [],
+            "2023-07-15T10:00:18Z",
+            3,
+        ),
+    ],
+)
+def test_deduplicate(
+    event_key: str,
+    ids_fetched_by_type: list[str],
+    time_from: str,
+    expected_results,
+):
+    """
+    Given:
+        - The events
+    When:
+        - run `is_duplicate` method
+    Then:
+        - Ensure that the events found with an ID that matches
+          the value in the `ids_fetched_by_type` list are removed
+    """
+    dedup = Deduplicate(ids_fetched_by_type, EventType.POLICY_LOGS)
+    events = []
+    for event in load_event_for_test(event_key):
+        if not dedup.is_duplicate(event, time_from):
+            events.append(event)
+    assert expected_results == len(events)
+
+
+@pytest.mark.parametrize(
+    "event",
+    [
+        {
+            "subject": "test",
+            "timestamp": "test"
+        },
+        {}
+    ]
+)
+def test_add_missing_fields_to_event(event: dict):
+    """
+    Givent:
+        - The event
```

Comment:
it's not just `the event`, it should be mentioned that it lacks some of the fields.